### ISSUE TITLE
[Component] [select-v2] select-v2 initStates 逻辑错误导致 modelValue 绑定值为对象时无法正确回显

### ISSUE THREAD

**Issue by LZQCN** [element-plus/element-plus#11089]
Bug Type: **`Component`**

## Environment

- Vue Version: `3.2.45`
- Element Plus Version: `2.2.27`
- Browser / OS: `Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/108.0.0.0 Safari/537.36`
- Build Tool: `Webpack`

### Related Component

- `el-select-v2`

### Reproduction Link

[Element Plus Playground](https://element-plus.run/#eyJBcHAudnVlIjoiPHNjcmlwdCBzZXR1cCBsYW5nPVwidHNcIj5cbmltcG9ydCB7IHJlZiwgdmVyc2lvbiBhcyB2dWVWZXJzaW9uIH0gZnJvbSAndnVlJ1xuaW1wb3J0IHsgdmVyc2lvbiBhcyBFcFZlcnNpb24sIEVsU2VsZWN0VjIgfSBmcm9tICdlbGVtZW50LXBsdXMnXG5pbXBvcnQgeyBFbGVtZW50UGx1cyB9IGZyb20gJ0BlbGVtZW50LXBsdXMvaWNvbnMtdnVlJ1xuXG5jb25zdCBsaXN0ID0gW1xuICAgIHsgaWQ6IDEsIG5hbWU6ICduYW1lMScgfSxcbiAgICB7IGlkOiAyLCBuYW1lOiAnbmFtZTInIH0sXG4gICAgeyBpZDogMywgbmFtZTogJ25hbWUzJyB9LFxuICAgIHsgaWQ6IDQsIG5hbWU6ICduYW1lNCcgfSxcbl1cblxuY29uc3QgdmFsdWUgPSByZWYobGlzdFswXSlcblxuY29uc3Qgb3B0aW9ucyA9IGxpc3QubWFwKG9iaiA9PiAoeyB2YWx1ZTogb2JqLCBsYWJlbDogb2JqLm5hbWUgfSkpXG48L3NjcmlwdD5cblxuPHRlbXBsYXRlPlxuICA8cD5cbiAgICA8ZWwtaWNvbiBjb2xvcj1cInZhcigtLWVsLWNvbG9yLXByaW1hcnkpXCI+PEVsZW1lbnRQbHVzIC8+PC9lbC1pY29uPlxuICAgIEVsZW1lbnQgUGx1cyB7eyBFcFZlcnNpb24gfX0gKyBWdWUge3sgdnVlVmVyc2lvbiB9fVxuICA8L3A+XG4gIDxFbFNlbGVjdFYyXG4gICAgdi1tb2RlbD1cInZhbHVlXCJcbiAgICA6b3B0aW9ucz1cIm9wdGlvbnNcIlxuICAgIHBsYWNlaG9sZGVyPVwiUGxlYXNlIHNlbGVjdFwiXG4gIC8+XG48L3RlbXBsYXRlPlxuIiwiaW1wb3J0X21hcC5qc29uIjoie1xuICBcImltcG9ydHNcIjoge31cbn0iLCJfbyI6e319)

### Steps to reproduce

如代码示例

### What is Expected?

modelValue为对象时能正常被回显

### What is actually happening?

[object Object]

### Additional comments

https://github.com/element-plus/element-plus/blob/74a1a16e386b843a8b9acb161d635dbee95216d0/packages/components/select-v2/src/useSelect.ts#L681

Bug Type: **`Component`**

## Environment

- Vue Version: `3.2.45`
- Element Plus Version: `2.2.27`
- Browser / OS: `Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/108.0.0.0 Safari/537.36`
- Build Tool: `Webpack`

### Related Component

- `el-select-v2`

### Reproduction Link

[Element Plus Playground](https://element-plus.run/#eyJBcHAudnVlIjoiPHNjcmlwdCBzZXR1cCBsYW5nPVwidHNcIj5cbmltcG9ydCB7IHJlZiwgdmVyc2lvbiBhcyB2dWVWZXJzaW9uIH0gZnJvbSAndnVlJ1xuaW1wb3J0IHsgdmVyc2lvbiBhcyBFcFZlcnNpb24sIEVsU2VsZWN0VjIgfSBmcm9tICdlbGVtZW50LXBsdXMnXG5pbXBvcnQgeyBFbGVtZW50UGx1cyB9IGZyb20gJ0BlbGVtZW50LXBsdXMvaWNvbnMtdnVlJ1xuXG5jb25zdCBsaXN0ID0gW1xuICAgIHsgaWQ6IDEsIG5hbWU6ICduYW1lMScgfSxcbiAgICB7IGlkOiAyLCBuYW1lOiAnbmFtZTInIH0sXG4gICAgeyBpZDogMywgbmFtZTogJ25hbWUzJyB9LFxuICAgIHsgaWQ6IDQsIG5hbWU6ICduYW1lNCcgfSxcbl1cblxuY29uc3QgdmFsdWUgPSByZWYobGlzdFswXSlcblxuY29uc3Qgb3B0aW9ucyA9IGxpc3QubWFwKG9iaiA9PiAoeyB2YWx1ZTogb2JqLCBsYWJlbDogb2JqLm5hbWUgfSkpXG48L3NjcmlwdD5cblxuPHRlbXBsYXRlPlxuICA8cD5cbiAgICA8ZWwtaWNvbiBjb2xvcj1cInZhcigtLWVsLWNvbG9yLXByaW1hcnkpXCI+PEVsZW1lbnRQbHVzIC8+PC9lbC1pY29uPlxuICAgIEVsZW1lbnQgUGx1cyB7eyBFcFZlcnNpb24gfX0gKyBWdWUge3sgdnVlVmVyc2lvbiB9fVxuICA8L3A+XG4gIDxFbFNlbGVjdFYyXG4gICAgdi1tb2RlbD1cInZhbHVlXCJcbiAgICA6b3B0aW9ucz1cIm9wdGlvbnNcIlxuICAgIHBsYWNlaG9sZGVyPVwiUGxlYXNlIHNlbGVjdFwiXG4gIC8+XG48L3RlbXBsYXRlPlxuIiwiaW1wb3J0X21hcC5qc29uIjoie1xuICBcImltcG9ydHNcIjoge31cbn0iLCJfbyI6e319)

### Steps to reproduce

如代码示例

### What is Expected?

modelValue为对象时能正常被回显

### What is actually happening?

[object Object]

### Additional comments

https://github.com/element-plus/element-plus/blob/74a1a16e386b843a8b9acb161d635dbee95216d0/packages/components/select-v2/src/useSelect.ts#L681

```
getValueKey(option) === getValueKey(props.modelValue)
```
这行代码明显有问题，它暗示 `option` 和 `props.modelValue` 具有相同的对象结构。假如
`valueKey` 是针对 `option` 设计的，则意味着最终输出时 `modelValue` 可能等于 `option` 的某个属性，显然两者不可能再具有相同的结构。

假如 `valueKey` 是针对 `modelValue` 设计的，那么在进行比较时应该对 `option.value` 使用 `valueKey`，而不是对 `option` 自身使用 `valueKey`，并且输出时不应再使用 `getValueKey(option)` 而应该直接输出 `option.value`。

**Comment by LZQCN** [comment]
亲测改为 getValueKey(option) === props.modelValue，能够正确回显对象类型绑定值了

**Comment by makedopamine** [comment]
Check this out. [playground](https://element-plus.run/#eyJBcHAudnVlIjoiPHNjcmlwdCBzZXR1cCBsYW5nPVwidHNcIj5cbmltcG9ydCB7IHJlZiwgdmVyc2lvbiBhcyB2dWVWZXJzaW9uIH0gZnJvbSAndnVlJ1xuaW1wb3J0IHsgdmVyc2lvbiBhcyBFcFZlcnNpb24sIEVsU2VsZWN0VjIgfSBmcm9tICdlbGVtZW50LXBsdXMnXG5pbXBvcnQgeyBFbGVtZW50UGx1cyB9IGZyb20gJ0BlbGVtZW50LXBsdXMvaWNvbnMtdnVlJ1xuXG5jb25zdCBsaXN0ID0gW1xuICAgIHsgaWQ6IDEsIG5hbWU6ICduYW1lMScgfSxcbiAgICB7IGlkOiAyLCBuYW1lOiAnbmFtZTInIH0sXG4gICAgeyBpZDogMywgbmFtZTogJ25hbWUzJyB9LFxuICAgIHsgaWQ6IDQsIG5hbWU6ICduYW1lNCcgfSxcbl1cblxuY29uc3QgdmFsdWUgPSByZWYoMSlcblxuY29uc3Qgb3B0aW9ucyA9IGxpc3QubWFwKG9iaiA9PiAoeyB2YWx1ZTogb2JqLCBsYWJlbDogb2JqLm5hbWUgfSkpXG48L3NjcmlwdD5cblxuPHRlbXBsYXRlPlxuICA8RWxTZWxlY3RWMlxuICAgIHYtbW9kZWw9XCJ2YWx1ZVwiXG4gICAgOm9wdGlvbnM9XCJvcHRpb25zXCJcbiAgICBwbGFjZWhvbGRlcj1cIlBsZWFzZSBzZWxlY3RcIlxuICAgIHZhbHVlLWtleT1cInZhbHVlLmlkXCJcbiAgLz5cbjwvdGVtcGxhdGU+XG4iLCJpbXBvcnRfbWFwLmpzb24iOiJ7XG4gIFwiaW1wb3J0c1wiOiB7fVxufSIsIl9vIjp7fX0=)

**Comment by LZQCN** [comment]
> Check this out. [playground](https://element-plus.run/#eyJBcHAudnVlIjoiPHNjcmlwdCBzZXR1cCBsYW5nPVwidHNcIj5cbmltcG9ydCB7IHJlZiwgdmVyc2lvbiBhcyB2dWVWZXJzaW9uIH0gZnJvbSAndnVlJ1xuaW1wb3J0IHsgdmVyc2lvbiBhcyBFcFZlcnNpb24sIEVsU2VsZWN0VjIgfSBmcm9tICdlbGVtZW50LXBsdXMnXG5pbXBvcnQgeyBFbGVtZW50UGx1cyB9IGZyb20gJ0BlbGVtZW50LXBsdXMvaWNvbnMtdnVlJ1xuXG5jb25zdCBsaXN0ID0gW1xuICAgIHsgaWQ6IDEsIG5hbWU6ICduYW1lMScgfSxcbiAgICB7IGlkOiAyLCBuYW1lOiAnbmFtZTInIH0sXG4gICAgeyBpZDogMywgbmFtZTogJ25hbWUzJyB9LFxuICAgIHsgaWQ6IDQsIG5hbWU6ICduYW1lNCcgfSxcbl1cblxuY29uc3QgdmFsdWUgPSByZWYoMSlcblxuY29uc3Qgb3B0aW9ucyA9IGxpc3QubWFwKG9iaiA9PiAoeyB2YWx1ZTogb2JqLCBsYWJlbDogb2JqLm5hbWUgfSkpXG48L3NjcmlwdD5cblxuPHRlbXBsYXRlPlxuICA8RWxTZWxlY3RWMlxuICAgIHYtbW9kZWw9XCJ2YWx1ZVwiXG4gICAgOm9wdGlvbnM9XCJvcHRpb25zXCJcbiAgICBwbGFjZWhvbGRlcj1cIlBsZWFzZSBzZWxlY3RcIlxuICAgIHZhbHVlLWtleT1cInZhbHVlLmlkXCJcbiAgLz5cbjwvdGVtcGxhdGU+XG4iLCJpbXBvcnRfbWFwLmpzb24iOiJ7XG4gIFwiaW1wb3J0c1wiOiB7fVxufSIsIl9vIjp7fX0=)

若按照此解决方案，则违背了 modelValue 绑定值类型为 Object 的初衷。

**Comment by LZQCN** [comment]
问题的矛盾点在于，如果
https://github.com/element-plus/element-plus/blob/74a1a16e386b843a8b9acb161d635dbee95216d0/packages/components/select-v2/src/useSelect.ts#L681
是正确的，则意味着
https://github.com/element-plus/element-plus/blob/74a1a16e386b843a8b9acb161d635dbee95216d0/packages/components/select-v2/src/useSelect.ts#L432
是错误的。

第一点，同时在 `option` 和 `modelValue` 上使用 `getValueKey()`，则暗示这两者具有相同的对象结构。
第二点， 这行代码 `update(getValueKey(option))` 企图使用 `valueKey` 从 `option` 中获取某个属性并输出，这将导致 `modelValue` 更新为 `option` 上的某个属性，显然与前一段逻辑相互矛盾。

**Comment by tolking** [comment]
Fixed in #9638, wait v2.3

**Comment by LZQCN** [comment]
> Fixed in #9638, wait v2.3

这个解决方案应该是有效的，我认为其中的这行代码是符合 valueKey 原始意图的。
```
getValueKey(option.value) === getValueKey(props.modelValue)
```
